src/lambda/commands/createNewSamApp.ts

```diff
@@ -61,20 +61,14 @@ export async function resumeCreateNewSamApp(
     let reason: CreateReason = 'complete'
     let samVersion: string | undefined
     const samInitState: SamInitState | undefined = activationReloadState.getSamInitState()
-
-    const pathToLaunch = samInitState?.path
-    if (!pathToLaunch) {
-        return
-    }
```

Comment:
Now the semantics are captured in `ext.didReload()`, so callers can make the decision instead of doing it implicitly in `resumeCreateNewSamApp()`. Easier to understand and would have avoided the bug.